compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -521,7 +521,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
     else {
       val cases =
         for ((child, idx) <- cls.children.zipWithIndex) yield {
-          val patType = if (child.isTerm) child.termRef else child.rawTypeRef
+          val patType = if (child.isTerm) child.termRef else child.reachableRawTypeRef
```

Comment:
I think this should be:
```suggestion
          val patType = if (child.isTerm) child.reachableTermRef else child.reachableRawTypeRef
```
to test this, add a `case object` in `object Internals`